frontend/src/components/DriverDetail/DriverDetail.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Vehicle } from '../../types';
+import styles from './driverDetail.module.css';
+import UserDetail, { UserContactInfo, OtherInfo } from '../UserDetail/UserDetail';
+import { phone, clock, wheel } from "./icons";
+
+type DriverDetailProps = {
+  // profilePic: string;
+  firstName: string;
+  lastName: string;
+  netId: string;
+  phone: string;
+  availability: string[][]
+  vehicle: Vehicle
+}
+
+const DriverDetail = (props: any) => {
+  console.log("driver detail running");
+  const driver: DriverDetailProps = props.location.state;
```

Comment:
Could you use the `useLocation` hook instead of accessing the location from the props?

Reply:
I tried to use it but when I did that I couldn't assign driver's type as DriverDetailProps. Instead I got an error saying "Type 'PoorMansUnknown' is not assignable to type 'DriverDetailProps'. Type 'undefined' is not assignable to type 'DriverDetailProps'." Is there a way to use useLocation while getting rid of this error?